x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticatorTests.java

```diff
@@ -951,7 +951,11 @@ private String buildAuthResponse(JWT idToken, @Nullable AccessToken accessToken,
         if (withAccessToken) {
             accessToken = new BearerAccessToken(Base64.getUrlEncoder().encodeToString(randomByteArrayOfLength(32)));
             AccessTokenHash expectedHash = AccessTokenHash.compute(accessToken, JWSAlgorithm.parse(alg));
-            idToken = JWTClaimsSet.parse(idToken.toJSONObject().appendField("at_hash", expectedHash.getValue()));
+            Map<String, Object> idTokenMap = idToken.toJSONObject();
+            idTokenMap.put("at_hash", expectedHash.getValue());
+            // This is necessary as if nonce claim is of type Nonce, the library won't take it into consideration when serializing the JWT
+            idTokenMap.put("nonce", idTokenMap.get("nonce").toString());
```

Comment:
> Could you please explain a bit more about this?

It looks like some internal change ( I assume the change from using json-smart's `JSONObject`s to `Map<String, Object>` in oauth2-oidc-sdk, means that ID token signature generation can only handle certain types. 

>  Is it not needed before the version upgrade? 

No

> Since it is necessary now, do we need somehow reflect this in production code?

No. We are an RP and thus we don't need to generate signed ID Tokens, so this behavior change doesn't affect us.